test/e2e/flowaggregator_test.go

```diff
@@ -498,6 +480,61 @@ func testHelper(t *testing.T, data *TestData, podAIPs, podBIPs, podCIPs, podDIPs
 			checkRecordsForFlows(t, data, podAIPs.ipv4.String(), svcC.Spec.ClusterIP, isServiceIPv6, false, true, false, false, checkBandwidth)
 		}
 	})
+
+	// RestartFlowAggregator tests the case, where Flow Aggregator stops, Flow Exporter should be able
+	// to detect and stop sending records until Flow Aggregator restarts.
+	t.Run("RestartFlowAggregator", func(t *testing.T) {
+		// In Kind cluster, there are issues with DNS name resolution on worker nodes.
+		// Flow Exporter needs to be restarted every time Flow Aggregator get restarts since it uses
+		// cluster IP as collector address instead of DNS name.
+		if testOptions.providerName == "kind" {
+			t.Skip("Skipping test because Flow Exporter needs to be restarted anyways in Kind cluster.")
+		}
+		err := testData.deleteFlowAggregator()
+		if err != nil {
+			t.Errorf("Error when deleting Flow Aggregator %v", err)
+		}
+		antreaPodName, err := data.getAntreaPodOnNode(workerNodeName(1))
+		if err != nil {
+			t.Errorf("Error when getting Antrea Pod name %v", err)
+		}
+		now := time.Now().Format(time.RFC3339)
+		errMsg := "Error when initializing flow exporter"
+		err = wait.PollImmediate(500*time.Millisecond, connCheckInterval, func() (bool, error) {
+			command := fmt.Sprintf("kubectl logs --pod-running-timeout=%v %s -n kube-system -c antrea-agent --since-time=%s", connCheckInterval.String(), antreaPodName, now)
+			rc, agentOutput, _, err := provider.RunCommandOnNode(controlPlaneNodeName(), command)
+			if err != nil || rc != 0 {
+				return false, err
+			}
+			return strings.Contains(agentOutput, errMsg), nil
+		})
+		assert.NoError(t, err)
+		testData.deployFlowAggregator(collectorAddr)
+	})
+
+	// ToExternalFlowsAfterRestart tests the export of IPFIX flow records when a source Pod
+	// sends traffic to an external IP after Flow Aggregator gets restarted.
+	t.Run("ToExternalFlowsAfterRestart", func(t *testing.T) {
```

Comment:
I am not a big fan of having a test depend on another test like this. IMO, it would be better to keep the `ToExternalFlows` test as it was, and have a dedicated test for restart that performs the necessary checks after restarting the aggregator

Reply:
I thought `ToExternalFlows` test was repeated after restart. I agree with Antonin that we should not move the test entirely. We should have this validation inside the restart test or have a separate `ToExternal` test repeating after restart.

Reply:
Got it. Updated the tests accordingly.